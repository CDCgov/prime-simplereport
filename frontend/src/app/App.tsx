import React, { useEffect } from "react";
import { gql, useQuery } from "@apollo/client";
import { useDispatch, connect } from "react-redux";
import { Redirect, Route, Switch } from "react-router-dom";
import { ApplicationInsights } from "@microsoft/applicationinsights-web";

import ProtectedRoute from "./commonComponents/ProtectedRoute";
import PrimeErrorBoundary from "./PrimeErrorBoundary";
import Header from "./commonComponents/Header";
import Page from "./commonComponents/Page/Page";
import LoginView from "./LoginView";
import { setInitialState } from "./store";
import TestResultsList from "./testResults/TestResultsList";
import TestQueueContainer from "./testQueue/TestQueueContainer";
import ManagePatientsContainer from "./patients/ManagePatientsContainer";
import EditPatientContainer from "./patients/EditPatientContainer";
import AddPatient from "./patients/AddPatient";
import AdminRoutes from "./admin/AdminRoutes";
import WithFacility from "./facilitySelect/WithFacility";
import { appPermissions } from "./permissions";
import Settings from "./Settings/Settings";
import { getAppInsights } from "./TelemetryService";
import VersionEnforcer from "./VersionEnforcer";

export const WHOAMI_QUERY = gql`
  query WhoAmI {
    whoami {
      id
      firstName
      middleName
      lastName
      suffix
      email
      isAdmin
      permissions
      roleDescription
      organization {
        name
        testingFacility {
          id
          name
        }
      }
    }
  }
`;

const App = () => {
  const appInsights = getAppInsights();

  const dispatch = useDispatch();
  const { data, loading, error } = useQuery(WHOAMI_QUERY, {
    fetchPolicy: "no-cache",
  });

  useEffect(() => {
    if (!data) return;

    dispatch(
      setInitialState({
        dataLoaded: true,
        organization: {
          name: data.whoami.organization?.name,
        },
        facilities: data.whoami.organization?.testingFacility || [],
        facility: null,
        user: {
          id: data.whoami.id,
          firstName: data.whoami.firstName,
          middleName: data.whoami.middleName,
          lastName: data.whoami.lastName,
          suffix: data.whoami.suffix,
          email: data.whoami.email,
          roleDescription: data.whoami.roleDescription,
          isAdmin: data.whoami.isAdmin,
          permissions: data.whoami.permissions,
        },
      })
    );
    // eslint-disable-next-line
  }, [data]);

  if (loading) {
    return <p>Loading account information...</p>;
  }

  if (error) {
    if (appInsights instanceof ApplicationInsights) {
      appInsights.trackException({ error });
    }
    return <p>Server connection error...</p>;
  }
  return (
    <PrimeErrorBoundary>
      <VersionEnforcer />
      <WithFacility>
        <Page>
          <Header />
          <Switch>
            <Route path="/login" component={LoginView} />
            <Route
              path="/queue"
              render={() => {
                return <TestQueueContainer />;
              }}
            />
            <Route
              path="/"
              exact
              render={({ location }) => (
                <Redirect
                  to={{
                    ...location,
                    pathname: data.whoami.isAdmin ? "/admin" : "/queue",
                  }}
                />
              )}
            />
            <ProtectedRoute
              path="/results/:page?"
              render={({ match }: any) => {
                return <TestResultsList page={match.params.page} />;
              }}
              requiredPermissions={appPermissions.results.canView}
              userPermissions={data.whoami.permissions}
            />
            <ProtectedRoute
              path={`/patients/:page?`}
              render={({ match }: any) => {
                return <ManagePatientsContainer page={match.params.page} />;
              }}
              requiredPermissions={appPermissions.people.canView}
              userPermissions={data.whoami.permissions}
            />
            <ProtectedRoute
              path={`/patient/:patientId`}
              render={({ match }: any) => (
                <EditPatientContainer patientId={match.params.patientId} />
              )}
              requiredPermissions={appPermissions.people.canEdit}
              userPermissions={data.whoami.permissions}
            />
            <ProtectedRoute
              path={`/add-patient/`}
              render={() => <AddPatient />}
              requiredPermissions={appPermissions.people.canEdit}
              userPermissions={data.whoami.permissions}
            />
            <ProtectedRoute
              path="/settings"
              component={Settings}
              requiredPermissions={appPermissions.settings.canView}
              userPermissions={data.whoami.permissions}
            />
            <Route
              path={"/admin"}
              render={({ match }) => (
                <AdminRoutes match={match} isAdmin={data.whoami.isAdmin} />
              )}
            />
          </Switch>
        </Page>
      </WithFacility>
    </PrimeErrorBoundary>
  );
};

export default connect()(App);
