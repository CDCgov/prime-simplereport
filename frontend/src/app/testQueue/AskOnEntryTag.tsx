import { useState, useEffect } from "react";

import { AoEAnswers } from "./AoEForm/AoEForm";

export const areAnswersComplete = (answers: AoEAnswers) => {
  if (!answers.noSymptoms) {
    const symptoms = JSON.parse(answers.symptoms);
    const filled = Object.values(symptoms).some((v) => String(v) === "true");
    if (!filled || !answers.symptomOnset) {
      return false;
    }
  }
  if (!answers.pregnancy) {
    return false;
  }
  return true;
};

interface Props {
  aoeAnswers: AoEAnswers;
}

const AskOnEntryTag: React.FC<Props> = ({ aoeAnswers }) => {
  const [answers, setAnswers] = useState(aoeAnswers);
  useEffect(() => {
    setAnswers(aoeAnswers);
  }, [aoeAnswers]);
  if (areAnswersComplete(answers)) {
    return <span className="usa-tag bg-green">COMPLETED</span>;
  } else {
    return <span className="usa-tag">PENDING</span>;
  }
};

export default AskOnEntryTag;
