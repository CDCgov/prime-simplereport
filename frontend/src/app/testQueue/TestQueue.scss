@use "../../scss/utilities";

.grid-container.queue-container-wide {
  max-width: utilities.$site-max-width;
}

.prime-test-result {
  .usa-radio {
    white-space: nowrap;
  }
}
