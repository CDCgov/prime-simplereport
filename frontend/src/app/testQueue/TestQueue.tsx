import React, { useEffect } from "react";
import { gql, useQuery } from "@apollo/client";
import { toast } from "react-toastify";
import { TransitionGroup, CSSTransition } from "react-transition-group";

import { showError } from "../utils";

import AddToQueueSearch from "./addToQueue/AddToQueueSearch";
import QueueItem from "./QueueItem";
import { TestQueuePerson, AoEAnswers } from "./AoEForm/AoEForm";

const pollInterval = 10_000;

const transitionDuration = 1000;
const onEmptyQueueEntering = (node: HTMLElement) => {
  node.style.opacity = "1";
};
const onEmptyQueueExiting = (node: HTMLElement) => {
  node.style.display = "none";
};

const onExiting = (node: HTMLElement) => {
  node.style.marginBottom = `-${node.offsetHeight}px`;
  node.style.opacity = "0";
  node.style.transition = `opacity ${transitionDuration}ms ease-out, margin ${transitionDuration}ms ease-out`;
  node.style.pointerEvents = "none";
};

const emptyQueueMessage = (
  <div className="grid-container prime-center card-container">
    <div className="grid-row">
      <div className="usa-card__body">
        <p>
          There are no tests running. Search for a person to start their test{" "}
        </p>
      </div>
    </div>
  </div>
);

export const queueQuery = gql`
  query GetFacilityQueue($facilityId: ID!) {
    queue(facilityId: $facilityId) {
      internalId
      pregnancy
      dateAdded
      symptoms
      symptomOnset
      noSymptoms
      firstTest
      priorTestDate
      priorTestType
      priorTestResult
      deviceType {
        internalId
        name
        model
        testLength
      }
      patient {
        internalId
        telephone
        birthDate
        firstName
        middleName
        lastName
        gender
        testResultDelivery
        preferredLanguage
      }
      result
      dateTested
      patientLink {
        internalId
      }
    }
    organization {
      testingFacility {
        id
        deviceTypes {
          internalId
          name
          model
          testLength
        }
        defaultDeviceType {
          internalId
          name
          model
          testLength
        }
      }
    }
  }
`;

interface Props {
  activeFacilityId: string;
}

interface QueueItemData extends AoEAnswers {
  internalId: string;
  dateAdded: string;
  deviceType: {
    internalId: string;
    testLength: number;
  };
  patient: TestQueuePerson;
  result: string;
  dateTested: string;
  patientLink: {
    internalId: string;
  };
}

const TestQueue: React.FC<Props> = ({ activeFacilityId }) => {
  const { data, loading, error, refetch, startPolling, stopPolling } = useQuery(
    queueQuery,
    {
      fetchPolicy: "no-cache",
      variables: {
        facilityId: activeFacilityId,
      },
    }
  );

  useEffect(() => {
    // Start polling on creation, stop on componenent teardown
    startPolling(pollInterval);
    return stopPolling;
  }, [startPolling, stopPolling]);

  if (error) {
    throw error;
  }
  if (loading) {
    return <p>Loading patients...</p>;
  }

  const facility = data.organization.testingFacility.find(
    (f: { id: string }) => f.id === activeFacilityId
  );
  if (!facility) {
    return <p>Facility not found</p>;
  }
  if (facility.deviceTypes.length === 0) {
    showError(
      toast,
      "This facility does not have any testing devices. Go into Settings -> Manage facilities and add a device."
    );
  }
  let shouldRenderQueue =
    data.queue.length > 0 && facility.deviceTypes.length > 0;

  const createQueueItems = (patientQueue: QueueItemData[]) => {
    const queue =
      shouldRenderQueue &&
      patientQueue.map(
        ({
          internalId,
          deviceType,
          patient,
          result,
          dateTested,
          patientLink,
          ...questions
        }) => {
          return (
            <CSSTransition
              key={internalId}
              onExiting={onExiting}
              timeout={transitionDuration}
            >
              <QueueItem
                internalId={internalId}
                patient={patient}
                askOnEntry={questions}
                selectedDeviceId={deviceType?.internalId || null}
                selectedDeviceTestLength={deviceType?.testLength || null}
                selectedTestResult={result}
                devices={facility.deviceTypes}
                defaultDevice={facility.defaultDeviceType}
                refetchQueue={refetch}
                facilityId={activeFacilityId}
                dateTestedProp={dateTested}
                patientLinkId={patientLink?.internalId || null}
              />
            </CSSTransition>
          );
        }
      );

    return (
      <TransitionGroup
        timeout={transitionDuration}
        component={null}
        unmountOnExit
      >
        {queue}
        {!shouldRenderQueue && (
          <CSSTransition
            key="empty-queue"
            onEntering={onEmptyQueueEntering}
            onExiting={onEmptyQueueExiting}
            timeout={transitionDuration}
          >
            {emptyQueueMessage}
          </CSSTransition>
        )}
      </TransitionGroup>
    );
  };

  const patientsInQueue = data.queue.map(
    (q: QueueItemData) => q.patient.internalId
  );

  return (
    <main className="prime-home">
      <div className="grid-container">
        <div className="position-relative">
          <AddToQueueSearch
            refetchQueue={refetch}
            facilityId={activeFacilityId}
            patientsInQueue={patientsInQueue}
          />
        </div>
        {createQueueItems(data.queue)}
      </div>
    </main>
  );
};

export default TestQueue;
