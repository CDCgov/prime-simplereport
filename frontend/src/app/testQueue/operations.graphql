
mutation RemovePatientFromQueue($patientId: ID!) {
    removePatientFromQueue(patientId: $patientId)
  }

mutation EditQueueItemMultiplex(
    $id: ID!
    $deviceId: String
    $deviceSpecimenType: ID
    $results: [MultiplexResultInput]
    $dateTested: DateTime
  ) {
    editQueueItemMultiplex(
      id: $id
      deviceId: $deviceId
      deviceSpecimenType: $deviceSpecimenType
      results: $results
      dateTested: $dateTested
    ) {
      results {
        disease {
          name
        }
        testResult
      }
      dateTested
      deviceType {
        internalId
        testLength
      }
      deviceSpecimenType {
        internalId
        deviceType {
          internalId
          testLength
        }
        specimenType {
          internalId
        }
      }
    }
  }

mutation SubmitTestResultMultiplex(
    $patientId: ID!
    $deviceId: String!
    $deviceSpecimenType: ID
    $results: [MultiplexResultInput]!
    $dateTested: DateTime
  ) {
    addTestResultMultiplex(
      patientId: $patientId
      deviceId: $deviceId
      deviceSpecimenType: $deviceSpecimenType
      results: $results
      dateTested: $dateTested
    ) {
      testResult {
        internalId
      }
      deliverySuccess
    }
  }
