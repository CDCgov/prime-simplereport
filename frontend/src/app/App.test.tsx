import { BrowserRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import createMockStore, { MockStoreEnhanced } from "redux-mock-store";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { render, screen } from "@testing-library/react";

import App, { WHOAMI_QUERY } from "./App";
import { queueQuery } from "./testQueue/TestQueue";

jest.mock("uuid");
jest.mock("./VersionService", () => ({
  VersionService: {
    enforce: jest.fn(),
  },
}));

const mockStore = createMockStore([]);
const mockDispatch = jest.fn();

const store = {
  dispatch: jest.fn(),
  organization: {
    name: "Organization Name",
  },
  user: {
    id: "05b2f71a-9392-442b-aab5-4eb550a864c0",
    firstName: "Bob",
    middleName: null,
    lastName: "Bobberoo",
    suffix: null,
    email: "bob@example.com",
    roleDescription: "Admin user",
    isAdmin: false,
    permissions: [
      "EDIT_PATIENT",
      "ARCHIVE_PATIENT",
      "READ_PATIENT_LIST",
      "EDIT_ORGANIZATION",
      "START_TEST",
      "EDIT_FACILITY",
      "ACCESS_ALL_FACILITIES",
      "READ_RESULT_LIST",
      "READ_ARCHIVED_PATIENT_LIST",
      "SUBMIT_TEST",
      "MANAGE_USERS",
      "SEARCH_PATIENTS",
      "UPDATE_TEST",
    ],
  },
  facilities: [
    {
      id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
      name: "Testing Site",
    },
  ],
  facility: {
    id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
    name: "Testing Site",
  },
};

store.dispatch = mockDispatch;

const WhoAmIQueryMock = {
  request: {
    query: WHOAMI_QUERY,
    fetchPolicy: "no-cache",
  },
  result: {
    data: {
      whoami: {
        id: "05b2f71a-9392-442b-aab5-4eb550a864c0",
        firstName: "Bob",
        middleName: null,
        lastName: "Bobberoo",
        suffix: null,
        email: "bob@example.com",
        isAdmin: false,
        permissions: [
          "EDIT_PATIENT",
          "ARCHIVE_PATIENT",
          "READ_PATIENT_LIST",
          "EDIT_ORGANIZATION",
          "START_TEST",
          "EDIT_FACILITY",
          "ACCESS_ALL_FACILITIES",
          "READ_RESULT_LIST",
          "READ_ARCHIVED_PATIENT_LIST",
          "SUBMIT_TEST",
          "MANAGE_USERS",
          "SEARCH_PATIENTS",
          "UPDATE_TEST",
        ],
        roleDescription: "Admin user",
        organization: {
          name: "Dis Organization",
          testingFacility: [
            {
              id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
              name: "Testing Site",
            },
          ],
        },
      },
    },
  },
};
const facilityQueryMock = {
  request: {
    query: queueQuery,
    fetchPolicy: "no-cache",
    variables: {
      facilityId: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
    },
  },
  result: {
    data: {
      queue: [],
      organization: {
        testingFacility: [
          {
            id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
            deviceTypes: [
              {
                internalId: "d70bb3b8-96bd-40d9-a3ce-b266a7edb91d",
                name: "Quidel Sofia 2",
                model: "Sofia 2 SARS Antigen FIA",
                testLength: 15,
              },
              {
                internalId: "5e44dcef-8cc6-44f4-a200-a5b8169ab60a",
                name: "LumiraDX",
                model: "LumiraDx SARS-CoV-2 Ag Test*",
                testLength: 15,
              },
            ],
            defaultDeviceType: {
              internalId: "5e44dcef-8cc6-44f4-a200-a5b8169ab60a",
              name: "LumiraDX",
              model: "LumiraDx SARS-CoV-2 Ag Test*",
              testLength: 15,
            },
          },
        ],
      },
    },
  },
};
const WhoAmIErrorQueryMock = {
  request: {
    query: WHOAMI_QUERY,
    fetchPolicy: "no-cache",
  },
  error: new Error("Server connection error"),
};
const renderApp = (
  newStore: MockStoreEnhanced<unknown, {}>,
  queryMocks: MockedResponse[]
) => {
  return render(
    <Provider store={newStore}>
      <MockedProvider mocks={queryMocks} addTypename={false}>
        <Router>
          <Route path="/" component={App} />
        </Router>
      </MockedProvider>
    </Provider>
  );
};

describe("App", () => {
  it("Render first loading screen", async () => {
    const mockedStore = mockStore({});
    renderApp(mockedStore, [WhoAmIQueryMock]);
    await screen.findByText("Loading account information...");
  });

  it("Render facility loading", async () => {
    const mockedStore = mockStore({ ...store });
    renderApp(mockedStore, [WhoAmIQueryMock, facilityQueryMock]);
    await screen.findByText("Loading facility information...");
  });

  it("Render main screen", async () => {
    const mockedStore = mockStore({ ...store, dataLoaded: true });
    const { container } = renderApp(mockedStore, [
      WhoAmIQueryMock,
      facilityQueryMock,
    ]);
    await screen.findByText("There are no tests running", { exact: false });
    expect(container).toMatchSnapshot();
  });
  it("should show error UI", async () => {
    const mockedStore = mockStore({ ...store, dataLoaded: true });
    const { container } = renderApp(mockedStore, [WhoAmIErrorQueryMock]);
    await screen.findByText("error", { exact: false });
    expect(container).toMatchSnapshot();
  });
});
