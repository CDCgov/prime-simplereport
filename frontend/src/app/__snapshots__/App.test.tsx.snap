// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render main screen 1`] = `
<div>
  <div
    class="App"
  >
    <div
      id="main-wrapper"
    >
      <div
        class="usa-banner"
      >
        <div
          class="usa-accordion"
        >
          <header
            class="usa-banner__header"
          >
            <div
              class="usa-banner__inner"
            >
              <div
                class="grid-col-auto"
              >
                <img
                  alt="U.S. flag"
                  class="usa-banner__header-flag"
                  src="us_flag_small.png"
                />
              </div>
              <div
                class="grid-col-fill tablet:grid-col-auto"
              >
                <p
                  class="usa-banner__header-text"
                >
                  An official website of the United States government
                </p>
                <p
                  aria-hidden="true"
                  class="usa-banner__header-action"
                >
                  Here’s how you know
                </p>
              </div>
              <button
                aria-controls="gov-banner"
                aria-expanded="false"
                class="usa-accordion__button usa-banner__button"
              >
                <span
                  class="usa-banner__button-text"
                >
                  Here’s how you know
                </span>
              </button>
            </div>
          </header>
        </div>
      </div>
      <header
        class="usa-header usa-header--basic"
      >
        <div
          class="usa-nav-container"
        >
          <div
            class="usa-navbar"
          >
            <div
              class="usa-logo"
              id="basic-logo"
            >
              <a
                aria-current="page"
                aria-label="Home"
                class="active"
                href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                title="Home"
              >
                <img
                  alt="{process.env.REACT_APP_TITLE}"
                  class="width-card desktop:width-full"
                  src="simplereport-logo-color.svg"
                />
              </a>
              <div
                class="prime-organization-name"
              >
                Organization Name
              </div>
            </div>
            <button
              class="usa-menu-btn"
            >
              Menu
            </button>
          </div>
          <nav
            aria-label="Primary navigation"
            class="usa-nav prime-nav desktop:display-none"
          >
            <button
              class="fa-layers fa-fw fa-2x usa-nav__close prime-nav-close-button"
              title="close menu"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-window-close fa-w-16 "
                data-icon="window-close"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <ul
              class="usa-nav__primary usa-accordion"
            >
              <li
                class="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
              >
                <a
                  aria-current="page"
                  class="prime-nav-link active-nav-item"
                  href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                  style="color: white;"
                >
                  Conduct tests
                </a>
              </li>
              <li
                class="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
              >
                <a
                  class="prime-nav-link"
                  href="/results?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  Results
                </a>
              </li>
              <li
                class="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
              >
                <a
                  class="prime-nav-link"
                  href="/patients?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  People
                </a>
              </li>
              <li
                class="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-user-circle fa-w-16 "
                  data-icon="user-circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style="fill: white;"
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                    fill="currentColor"
                  />
                </svg>
              </li>
              <li
                class="usa-nav__primary-item usa-sidenav prime-staff-infobox-sidemenu prime-settings-hidden"
              >
                <ul
                  class="usa-sidenav__sublist prime-sidenav_inset"
                >
                  <li
                    class="usa-sidenav__item span-full-name"
                  >
                    Bob Bobberoo
                  </li>
                  <li
                    class="usa-sidenav__item"
                  >
                    <span>
                      <strong>
                        Role: 
                      </strong>
                      Admin
                    </span>
                  </li>
                  <li
                    class="usa-sidenav__item"
                  >
                    Testing Site
                  </li>
                </ul>
              </li>
              <div>
                <div
                  class="navlink__support"
                >
                  <a
                    href="https://simplereport.gov/support"
                    target="none"
                  >
                    Support
                  </a>
                </div>
                <button
                  class="usa-button usa-button--unstyled"
                  type="button"
                >
                  Log out
                </button>
              </div>
              <li
                class="usa-nav__primary-item prime-settings-hidden"
              >
                <a
                  href="/settings?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-cog fa-w-16 "
                    data-icon="cog"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                      fill="currentColor"
                    />
                  </svg>
                   Settings
                </a>
              </li>
            </ul>
          </nav>
          <nav
            aria-label="Primary navigation"
            class="usa-nav prime-nav"
          >
            <ul
              class="usa-nav__primary usa-accordion"
            >
              <li
                class="usa-nav__primary-item"
              >
                <a
                  aria-current="page"
                  class="prime-nav-link active-nav-item"
                  href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                  id="conduct-test-nav-link"
                  style="color: white;"
                >
                  Conduct tests
                </a>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <a
                  class="prime-nav-link"
                  href="/results?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                  id="results-nav-link"
                >
                  Results
                </a>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <a
                  class="prime-nav-link"
                  href="/patients?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                  id="patient-nav-link"
                >
                  People
                </a>
              </li>
            </ul>
            <div
              class="prime-facility-select"
            >
              <div
                class="usa-form-group prime-dropdown "
              >
                <select
                  aria-required="false"
                  class="usa-select"
                  id="1"
                >
                  <option
                    value="fec4de56-f4cc-4c61-b3d5-76869ca71296"
                  >
                    Testing Site
                  </option>
                </select>
              </div>
            </div>
            <ul
              class="usa-nav__primary usa-accordion"
            >
              <li
                class="usa-nav__primary-item nav__primary-item-icon"
              >
                <a
                  data-testid="user-button"
                  href="/queue#?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-user-circle fa-w-16 "
                    data-icon="user-circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 496 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
                <div
                  aria-label="Primary navigation"
                  class="shadow-3 prime-staff-infobox"
                >
                  <ul
                    class="usa-sidenav__sublist"
                  >
                    <li
                      class="usa-sidenav__item span-full-name"
                    >
                      Bob Bobberoo
                    </li>
                    <li
                      class="usa-sidenav__item"
                    >
                      <span>
                        <strong>
                          Role: 
                        </strong>
                        Admin
                      </span>
                    </li>
                    <li
                      class="usa-sidenav__item"
                    >
                      Testing Site
                    </li>
                    <li
                      class="usa-sidenav__item navlink__support"
                    >
                      <a
                        data-testid="support-link"
                        href="https://simplereport.gov/support"
                        target="none"
                      >
                        Support
                      </a>
                    </li>
                    <li
                      class="usa-sidenav__item margin-top-2"
                    >
                      <button
                        class="usa-button usa-button--unstyled"
                        type="button"
                      >
                         Log out
                      </button>
                    </li>
                  </ul>
                </div>
              </li>
              <li
                class="usa-nav__primary-item nav__primary-item-icon"
              >
                <a
                  href="/settings?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-cog fa-w-16 "
                    data-icon="cog"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <main
        class="prime-home"
      >
        <div
          class="grid-container"
        >
          <div
            class="position-relative"
          >
            <div
              class="prime-search-container"
            >
              <form
                class="usa-search usa-search--small prime-search-input display-inline-block"
                role="search"
              >
                <label
                  class="usa-sr-only"
                  for="search-field-small"
                >
                  Search
                </label>
                <div>
                  <input
                    autocomplete="off"
                    class="usa-input"
                    id="search-field-small"
                    name="search"
                    placeholder="Search for a person to start their test"
                    type="search"
                    value=""
                  />
                  <button
                    class="usa-button"
                    disabled=""
                    type="submit"
                  >
                    <span
                      class="usa-sr-only"
                    >
                      Search
                    </span>
                  </button>
                </div>
              </form>
            </div>
          </div>
          <div
            class="grid-container prime-center card-container"
          >
            <div
              class="grid-row"
            >
              <div
                class="usa-card__body"
              >
                <p>
                  There are no tests running. Search for a person to start their test
                   
                </p>
              </div>
            </div>
          </div>
        </div>
      </main>
      <div
        class="Toastify"
      />
    </div>
  </div>
</div>
`;

exports[`App should show error UI 1`] = `
<div>
  <p>
    Server connection error...
  </p>
</div>
`;
